src/views/components/map/Map.web.js

```diff
@@ -1,16 +1,510 @@
 import React, { Component } from 'react';
-import { Box, Text } from '../../components';
+import { string, object, array, number, bool, any, func } from 'prop-types';
+import config from '../../../config';
 
+/* global google */
 class Map extends Component {
-  render() {
-    return (
-      <Box>
-        <Text>
-          TODO Map
-        </Text>
-      </Box>
-    );
-  }
+    static defaultProps = {
+      controls: false,
+      zoom: 12,
+      center: { lat: -33.8688, lng: 151.2093 },
+      markers: [],
+      routes: [],
+      icon: 'https://i.imgur.com/unMXE8B.png',
+      iconClick: 'https://i.imgur.com/XiZYxed.png',
+      mapRouteIcon: 'https://i.imgur.com/Zyinht5.png',
+      suppressMarkers: true,
+      polygons: [],
+      polylines: [],
+      height: 300,
+      width: 400,
+    }
+
+    static propTypes = {
+      style: object,
+      suppressMarkers: bool,
+      controls: bool,
+      zoom: number,
+      maxZoom: number,
+      center: any,
+      markers: array,
+      routes: array,
+      polygons: array,
+      polylines: array,
+      icon: string,
+      iconClick: string,
+      lat: any,
+      lng: any,
+      onClick: func,
+      mapRouteIcon: string,
+      height: string, 
+      width: string,
+
+    }
+
+    state = {
+      error: null,
+    };
+
+    UNSAFE_componentWillMount() { 
+      this.setupGoogleApi();
+    }
+    
+    componentWillUnmount() {
+      if ( this.map ) {
+        this.map = null;
+      }
+    }
+
+    setupGoogleApi = () => {
+      const googleScript = document.getElementById( 'google-api' );
+
+      if ( !googleScript ) {
+        const script = document.createElement( 'script' );
+
+        script.src =
+          `https://maps.googleapis.com/maps/api/js?key=${config.google.apiKey}`;
+  
+        script.async = true;
+        script.id = 'google-api';
+        document.body.appendChild( script );
+      }
+    };
+
+    setup = () => {
+      if ( typeof google === 'object' ) {
```

Comment:
Can we please use our internal `isObject` inside our `src/utils` directory? Simply checking if `google` is a `typeof` object still returns true if `google` is `null` or an array, so I consider it as unsafe practice.